dedupe/api.py

```diff
@@ -175,7 +174,6 @@ def partition(
         clusters = self.cluster(pair_scores, threshold)
         clusters = self._add_singletons(data, clusters)
         clusters = list(clusters)
```

Comment:
Now that we have auto-cleanup, perhaps we don't want to cast this to a list? I think then this entire method would be out-of-core, right?

Same for casting to list in join()

Reply:
that's right. i think we we should put in a warning for a few releases though because some folks are likely depending on it being a list instead of a generator

Reply:
do we want to do what Gazeteer.search() does with the `generator: bool` parameter?

```python
def score(.... generator: bool = False):
    ...
    if not generator:
        warn("this deprecated")
        return list(scores)
    else:
        yield from scores
```